ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/NetworkProtocolVersion.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE TypeFamilies #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Ouroboros.Consensus.Shelley.Ledger.NetworkProtocolVersion (
+    ShelleyNodeToClientVersion(..)
+  ) where
+
+import           Data.List.NonEmpty (NonEmpty (..))
+
+import qualified Ouroboros.Network.NodeToClient as N
+import qualified Ouroboros.Network.NodeToNode as N
+
+import           Ouroboros.Consensus.Node.NetworkProtocolVersion
+
+import           Ouroboros.Consensus.Shelley.Ledger.Block
+
+data ShelleyNodeToNodeVersion = ShelleyNodeToNodeVersion1
+  deriving (Show, Eq, Ord, Enum, Bounded)
+
+data ShelleyNodeToClientVersion = ShelleyNodeToClientVersion1
+  deriving (Show, Eq, Ord, Enum, Bounded)
+
+instance HasNetworkProtocolVersion (ShelleyBlock c) where
+  type NodeToNodeVersion   (ShelleyBlock c) = ShelleyNodeToNodeVersion
+  type NodeToClientVersion (ShelleyBlock c) = ShelleyNodeToClientVersion
+
+  supportedNodeToNodeVersions   _ = ShelleyNodeToNodeVersion1
+                                  :| []
+  supportedNodeToClientVersions _ = ShelleyNodeToClientVersion1
+                                  :| []
+
+  mostRecentNodeToNodeVersion   _ = ShelleyNodeToNodeVersion1
+  mostRecentNodeToClientVersion _ = ShelleyNodeToClientVersion1
+
+  nodeToNodeProtocolVersion _ ShelleyNodeToNodeVersion1 = N.NodeToNodeV_1
+
+  nodeToClientProtocolVersion _ ShelleyNodeToClientVersion1 = N.NodeToClientV_2
```

Comment:
Document that Shelley supports V_2, i.e., the enabled local state query protocol, from the start.